### PR TITLE
manifest: reduce CRD size by removing some descriptions

### PR DESCRIPTION
**Pull request by BlaineEXE**
Reduce CRD size by removing some descriptions for things that (a) are long, (b) are repeated, and (c) are commong kubernetes types.

Remove the `Status` field from VolumeClaimTemplates (a.k.a. PVCs) in types.go to remove all unnecessary status-related fields.

Use `yq` to go through generated `crds.yaml` to strip out descriptions for well-known k8s types that are long and repeated where we can't remove them from types.go. This is mostly placement descriptions and volumeSource descriptions.

The overall `crds.yaml` file is reduced to less than 60% of its original size.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: deploy/examples/crds.yaml [hunk 1]
```diff
@@ -5004,109 +4146,28 @@ spec:
...
+                                description: 'volumeAttributesClassName may be used to set the VolumeAttributesClass used by this claim. If specified, the CSI driver will create or update the volume with the attributes defined in the corresponding VolumeAttributesClass. This has a different purpose than storageClassName, it can be changed after the claim is created. An empty string value means that no VolumeAttributesClass will be applied to the claim but it''s not allowed to reset this field to empty string once it is set. If unspecified and the PersistentVolumeClaim is unbound, the default VolumeAttributesClass will be set by the persistentvolume controller if it exists. If the resource referred to by volumeAttributesClass does not exist, this PersistentVolumeClaim will be set to a Pending state, as reflected by the modifyVolumeStatus field, until such as a resource exists. More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#volumeattributesclass (Alpha) Using this field requires the VolumeAttributesClass feature gate to be enabled.'
```
Comment (BlaineEXE) [inline deploy/examples/crds.yaml:4157]: IMO, this is an example of when we do want to keep the long description. This is a Rook type, and the description might be most effective for users at full length.

File: deploy/examples/crds.yaml [hunk 2]
```diff
@@ -5982,7 +4905,7 @@ kind: CustomResourceDefinition
 metadata:
   annotations:
     controller-gen.kubebuilder.io/version: v0.11.3
-  creationTimestamp: null
+  creationTimestamp: {}
```
Comment (BlaineEXE) [inline deploy/examples/crds.yaml:4908]: I'm seeing `null` converted to `{}` here. I guess this is probably `yq`, but I'm not sure how to make it stop this.

Comment (BlaineEXE) [reply deploy/examples/crds.yaml]: It did cause an issue in the build, but I also think the field is unnecessary in CRD manifests, so I'm trying to just remove it.

Comment (subhamkrai) [reply deploy/examples/crds.yaml]: seems like ci is still not happy, I see logs in canary-tests/canary
```
2024-02-15 18:33:01.011255 D | op-mon: Released lock for mon orchestration
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x15992dd]

goroutine 672 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.1/pkg/internal/controller/controller.go:116 +0x1e5
panic({0x21241e0?, 0x3da1d80?})
	/opt/hostedtoolcache/go/1.21.7/x64/src/runtime/panic.go:914 +0x21f
github.com/rook/rook/pkg/apis/ceph.rook.io/v1.(*VolumeClaimTemplate).ToPVC(0xf32f77273dc4b1af?)
	/home/runner/work/rook/rook/pkg/apis/ceph.rook.io/v1/volume.go:54 +0x1d
````

File: pkg/apis/ceph.rook.io/v1/types.go [hunk 3]
```diff
@@ -598,7 +598,20 @@ type MonSpec struct {
 	// VolumeClaimTemplate is the PVC definition
 	// +kubebuilder:pruning:PreserveUnknownFields
 	// +optional
-	VolumeClaimTemplate *v1.PersistentVolumeClaim `json:"volumeClaimTemplate,omitempty"`
+	VolumeClaimTemplate *VolumeClaimTemplate `json:"volumeClaimTemplate,omitempty"`
```
Comment (subhamkrai) [inline pkg/apis/ceph.rook.io/v1/types.go:601]:
```suggestion
	VolumeClaimTemplate *VolumeClaimTemplate `json:"volumeClaimTemplate,omitempty"`
```

why we are not using default ones?

Comment (BlaineEXE) [reply pkg/apis/ceph.rook.io/v1/types.go:601]: Do you mean PVC? The reason is that this includes a `Status` field that we can't make use of in our CRD, and it generates into several CRDs. We can remove it to save CRD space.

Comment (subhamkrai) [reply pkg/apis/ceph.rook.io/v1/types.go:601]: got it.

**Pull request closed without merging**